Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -594,44 +611,49 @@ bool PcapLiveDevice::sendPacket(const uint8_t* packetData, int packetDataLength)
 	return true;
 }
 
-bool PcapLiveDevice::sendPacket(Packet* packet)
+bool PcapLiveDevice::sendPacket(Packet* packet, bool checkMtu)
 {
 	RawPacket* rawPacket = packet->getRawPacket();
-	return sendPacket(*rawPacket);
+	if (checkMtu)
+	{
+		int packetPayloadLength = (int)packet->getFirstLayer()->getLayerPayloadSize();

```

Comment:
this line assumes the first layer is a data link layer, but it doesn't have to be like that: https://github.com/seladb/PcapPlusPlus/blob/master/Packet%2B%2B/src/Packet.cpp#L713-L744

I'd suggest checking here if: `packet->getFirstLayer()->getOsiModelLayer() == pcpp::OsiModelDataLinkLayer`

Reply:
Thanks. Will add this.

I can add a check for if the first layer is `pcpp::OsiModelNetworkLayer` as well. In that instance, the MTU should just be the length of the packet.

Reply:
yes, that makes sense